Raspberry Pi/PyCanRecdBusSend.py

```diff
@@ -0,0 +1,98 @@
+import can
+from piracer.vehicles import PiRacerStandard
+from pydbus import SessionBus
+from gi.repository import GLib
+import concurrent.futures
+
+global BatteryLevel, Rpm
+
+#########Battery Level in Percent ######################################################################
+def BatteryLevelinPer(x): #Calculate Voltage to Percent
+    return -25.6269 * x**3 + 887.9706 * x**2 - 10180.5203 * x + 38661.8396
+
+class RollingAverage: #Averaging the last 5 Voltages
+    def __init__(self, window_size=5):
+        self.window_size = window_size
+        self.measurements = []
+
+    def add_measurement(self, value):
+        self.measurements.append(value)
+        if len(self.measurements) > self.window_size:
+            self.measurements = self.measurements[-self.window_size:]
+
+    def get_average(self):
+        if not self.measurements:
+            return None  # Return None if no measurements are available
+        return sum(self.measurements) / len(self.measurements)
+
+def getBatteryLevel():
+    rolling_average = RollingAverage()
+    #The following lines are important to get Voltage, Average and Transform into Percent: From battery_voltage to batteryLevel
+    battery_voltage = piracer.get_battery_voltage() #important to get Voltage
+    measurement = battery_voltage           #Voltage as Measurement
+    rolling_average.add_measurement(measurement)    #Averaging the last 5 Voltages
+    average = rolling_average.get_average()
+    BatteryLevel = str(BatteryLevelinPer(average))       #Calculate Voltage to Percent
+
+#########RPM from CAN Bus######################################################################
+# Create a CAN bus channel
+can_bus = can.interface.Bus(interface='socketcan', channel='can0', bitrate=1000000)
+
+def getRPM(self):
```

Comment:
@NikDoh Have you merged both scripts i.e. RPM and Battery? If yes, have you checked whether it is properly working?

Reply:
ok i see its the wrong file i will upload the right one next week.